spk/sonarr3/src/service-setup.sh

```diff
@@ -0,0 +1,92 @@
+if [ "$SYNOPKG_DSM_ARCH" == "88f6281" ] || [ "$SYNOPKG_DSM_ARCH" == "88f6282" ]; then
+    MONO="MONO_ENV_OPTIONS='-O=-aot,-float32' ${MONO}"
+fi
+
+# for DSM < 7 only:
+GROUP="sc-download"
+LEGACY_GROUP="sc-media"
+
+SERVICE_COMMAND="env PATH=${MONO_PATH}:${PATH} HOME=${HOME_DIR} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${MONO} ${SONARR}"
+SVC_BACKGROUND=y
+SVC_WAIT_TIMEOUT=90
+
+validate_preinst ()
+{
+    # use install_log to write to installer log file.
+    install_log "validate_preinst ${SYNOPKG_PKG_STATUS}"
+
+    # check if the installed distribution is a legacy version
+    if [ -f "${SYNOPKG_PKGDEST}/share/NzbDrone/NzbDrone.exe" ]; then
+        # v2 installed
+        exit 1
+    fi
+}
+
+service_postinst ()
+{
+    echo "Set update required"
+    # Make Sonarr do an update check on start to avoid possible Sonarr
+    # downgrade when synocommunity package is updated
+    touch "${SONARR_CONFIG_DIR}/update_required"
+    
+    if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -lt 7 ]; then
+        set_unix_permissions "${CONFIG_DIR}"
+    fi
+}
+
+service_preupgrade ()
+{
+    if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -ge 7 ]; then
+        # ensure config is in @appdata folder
+        if [ -d "${LEGACY_CONFIG_DIR}" ]; then
+            if [ "$(realpath "${LEGACY_CONFIG_DIR}")" != "$(realpath "${CONFIG_DIR}")" ]; then
+                echo "Move ${LEGACY_CONFIG_DIR} to ${CONFIG_DIR}"
+                mv "${LEGACY_CONFIG_DIR}" "${CONFIG_DIR}" 2>&1
+            fi
+        fi
+    fi
+
+    # never update Sonarr distribution, use internal updater only
+    [ -d "${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup" ] && rm -rf "${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup"
+    echo "Backup existing distribution to ${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup"
+    mkdir -p "${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup" 2>&1
+    rsync -aX "${SYNOPKG_PKGDEST}/share" "${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup/" 2>&1
+}
+
+service_postupgrade ()
+{
+    # restore Sonarr distribution
+    if [ -d "${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup/share" ]; then
+        echo "Restore previous distribution from ${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup"
+        rm -rf "${SYNOPKG_PKGDEST}/share/Sonarr/bin" 2>&1
+        # prevent overwrite of updated package_info
+        rsync -aX --exclude=package_info "${SYNOPKG_TEMP_UPGRADE_FOLDER}/backup/share/" "${SYNOPKG_PKGDEST}/share" 2>&1
+    fi
+
+    if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -lt 7 ]; then
+        set_unix_permissions "${SYNOPKG_PKGDEST}/share"
+        # If backup was created before new-style packages
+        # new updates/backups will fail due to permissions (see #3185)
+        if [ -d "/tmp/nzbdrone_backup" ] || [ -d "/tmp/nzbdrone_update" ] || [ -d "/tmp/sonarr_backup" ] || [ -d "/tmp/sonarr_update" ]; then
+            set_unix_permissions "/tmp/nzbdrone_backup"
+            set_unix_permissions "/tmp/nzbdrone_update"
+            set_unix_permissions "/tmp/sonarr_backup"
+            set_unix_permissions "/tmp/sonarr_update"
+        fi
```

Comment:
we probably must move these folders to the new TMPDIR

An updated comment would be like:
# Move backup folders created before alternate TMPDIR is used

Reply:
I'm not sure what to do here. Should we be checking for these folders and just dumping them? Because wouldn't the backup folders adhere to the new locations created in `service_preupgrade` with the fix for the `USE_ALTERNATE_TMPDIR` you gave above?

Reply:
@hgy59, I've pushed a new commit. Let me know if this is what you were considering as a solution.

Reply:
> @hgy59, I've pushed a new commit. Let me know if this is what you were considering as a solution.

@mreid-tt 
to answer your question I have added some commits.

My finding was, that those folders in /tmp are really temporary and just can be removed.
- The *_backup folder is temp. for the backup. The final backup file(s) (*.zip) are stored in the config dir under Backup and are available for download in the sonarr web UI.
- The *_update folder is used to download and extract updates. It is obsolete after the update

And I reverted to mono>=5.8 since there are issues with 5.20 and aarch64 on DSM 6 (built with DSM 6 toolchain).